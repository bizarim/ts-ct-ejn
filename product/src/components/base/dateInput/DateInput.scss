.date-input {
    
    input {
        height: calc(2.35em);
        border: 1px solid var(--sub-2-border-color);
        padding: .6875rem .75rem;
        border-radius: 4px;
        box-shadow: none;
        font-size: 1rem;

        &:focus {
            outline: 0;
        }
    }

    input[type="date"] {
        background: #fff url(https://cdn1.iconfinder.com/data/icons/cc_mono_icon_set/blacks/16x16/calendar_2.png) 93% 50% no-repeat;
    }
    input[type="date"]::-webkit-inner-spin-button {
        display: none;
    }
    input[type="date"]::-webkit-calendar-picker-indicator {
        opacity: 0;
    }
}
