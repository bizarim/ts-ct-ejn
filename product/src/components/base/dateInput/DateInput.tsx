import React from 'react';

export interface DateInputProps {
    text: string;
    onHandle(text: string): void;
}
interface State {
    value: string;
}

export class DateInput extends React.Component<DateInputProps, State> {
    public state = {
        value: this.props.text,
    };
    public render() {
        const { value } = this.state;
        return (
            <div className="date-input">
                <input
                    name="input-date"
                    type="date"
                    value={value}
                    onChange={this.onChange}
                />
            </div>
        );
    }

    private onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const value = e.target.value;
        const { onHandle } = this.props;
        this.setState({ value: value });

        if (onHandle)
            onHandle(value);
    };
}