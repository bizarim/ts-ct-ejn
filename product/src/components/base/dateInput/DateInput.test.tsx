import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import { DateInputProps, DateInput } from './DateInput';

configure({ adapter: new Adapter() });

const defaultProps: DateInputProps = {
    text: 'test',
    onHandle: jest.fn(),
};

const setup = (props: Partial<DateInputProps> = {}) =>
    shallow(<DateInput {...{ ...defaultProps, ...props }} />);

describe('DateInput', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('should have correct className date-input', () => {
        const wrapper = setup();
        expect(wrapper.hasClass('date-input')).toBeTruthy();
        // todo
        // find children
    });

    it('should selected DateInput', () => {
        const onChange = jest.fn();
        const props: DateInputProps = {
            text: 'XXXXX',
            onHandle: onChange,
        };
        const wrapper = setup(props);
        const dateInput = wrapper.find('input');
        dateInput.simulate('change', { target: { value : 'YYY'} });
        expect(onChange).toHaveBeenCalledWith('YYY');
    });


});
