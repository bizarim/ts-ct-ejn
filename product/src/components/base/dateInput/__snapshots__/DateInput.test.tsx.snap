// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput should render 1`] = `ShallowWrapper {}`;
